compiler-core/src/ast.rs

```diff
@@ -747,6 +747,40 @@ impl BinOp {
             Self::Concatenate => "<>",
         }
     }
+
+    pub fn can_be_grouped_with(self, other: BinOp) -> bool {
+        let is_boolean_binop = |binop| matches!(binop, BinOp::And | BinOp::Or);
+        let is_int_comparison = |binop| {
+            matches!(
```

Comment:
Could you convert these to match expressions that list every possible value? With this macro and wildcards the compiler doesn't tell you if a new variant is added, which can result in bugs as logic doesn't get updated.

Might be good to extract them to methods too.